tests/tests/test-xcm/test-xcmv3-new-instructions.ts

```diff
@@ -0,0 +1,382 @@
+      const xcmMessage = new XcmFragment(dotAsset)
+        .withdraw_asset()
+        .buy_execution()
+        .clear_transact_status()
+        .as_v3();
+
+      // Mock the reception of the xcm message
+      await injectHrmpMessage(context, paraId, {
+        type: "XcmVersionedXcm",
+        payload: xcmMessage,
+      } as RawXcmMessage);
+      await context.createBlock();
+
+      // Search for Transport error
+      const records = (await context.polkadotApi.query.system.events()) as any;
+      const events = records.filter(
+        ({ event }) => event.section == "xcmpQueue" && event.method == "Success"
+      );
+      expect(events).to.have.lengthOf(1);
+    });
+
+    it("Should execute SetFeesMode", async function () {
+      const xcmMessage = new XcmFragment(dotAsset)
+        .withdraw_asset()
+        .buy_execution()
+        .set_fees_mode()
+        .as_v3();
+
+      // Mock the reception of the xcm message
+      await injectHrmpMessage(context, paraId, {
+        type: "XcmVersionedXcm",
+        payload: xcmMessage,
+      } as RawXcmMessage);
+      await context.createBlock();
+
+      // Search for Success
+      const records = (await context.polkadotApi.query.system.events()) as any;
+      const events = records.filter(
+        ({ event }) => event.section == "xcmpQueue" && event.method == "Success"
+      );
+      expect(events).to.have.lengthOf(1);
+    });
+
+    it("Should execute SetTopic", async function () {
+      const xcmMessage = new XcmFragment(dotAsset)
+        .withdraw_asset()
+        .buy_execution()
+        // SetTopic expects an array of 32 bytes
+        .set_topic([
+          122, 22, 113, 160, 34, 76, 137, 39, 176, 143, 151, 128, 39, 213, 134, 171, 104, 104, 222,
+          13, 49, 187, 91, 201, 86, 182, 37, 206, 210, 171, 24, 196,
+        ])
+        .as_v3();
+
+      // Mock the reception of the xcm message
+      await injectHrmpMessage(context, paraId, {
+        type: "XcmVersionedXcm",
+        payload: xcmMessage,
+      } as RawXcmMessage);
+      await context.createBlock();
+
+      // Search for UnknownClaim error
+      const records = (await context.polkadotApi.query.system.events()) as any;
+      const events = records.filter(
+        ({ event }) => event.section == "xcmpQueue" && event.method == "Success"
+      );
+      expect(events).to.have.lengthOf(1);
+    });
+
+    it("Should execute ReportHolding (Transport error)", async function () {
+      const xcmMessage = new XcmFragment(dotAsset)
+        .withdraw_asset()
+        .buy_execution()
+        .report_holding(1000)
+        .as_v3();
+
+      // Mock the reception of the xcm message
+      await injectHrmpMessage(context, paraId, {
+        type: "XcmVersionedXcm",
+        payload: xcmMessage,
+      } as RawXcmMessage);
+      await context.createBlock();
+
+      const records = (await context.polkadotApi.query.system.events()) as any;
+      const events = records.filter(
```

Comment:
same comment as in the `tests/tests/test-xcm/test-xcmv3-max-weight-instructions.ts`